src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -104,12 +104,22 @@ export function ActiveWeb3Provider({
     [jsonRpcUrlMap, defaultChainId]
   )
 
-  connections = [metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection]
-  if (integratorConnection) connections = [integratorConnection, ...connections]
+  const key = useRef(0)
+  connections = useMemo(() => {
+    key.current += 1
+    let connections = [metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection]
```

Comment:
I think this is more cleanly expressed as

    return [integratorConnection, metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection].filter(Boolean)